morpheus/_lib/include/morpheus/messages/meta.hpp

```diff
@@ -101,6 +99,43 @@ class MessageMeta
     std::shared_ptr<IDataTable> m_data;
 };
 
+/**
+ * @brief Operates similarly to MessageMeta, except it applies a filter on the columns and rows. Used by Serialization
+ * to filter columns without copying the entire DataFrame
+ *
+ */
+class SlicedMessageMeta : public MessageMeta
+{
+  public:
+    SlicedMessageMeta(std::shared_ptr<MessageMeta> other,
+                      cudf::size_type start            = 0,
+                      cudf::size_type stop             = -1,
+                      std::vector<std::string> columns = {});
+
+    TableInfo get_info() const override;
+
+    MutableTableInfo get_mutable_info() const override;
+
+  private:
+    cudf::size_type m_start{0};
+    cudf::size_type m_stop{-1};
+    std::vector<std::string> m_column_names;
+};
+
+/****** Python Interface **************************/
+class MutableCtxMgr : public std::enable_shared_from_this<MutableCtxMgr>
+{
+  public:
+    MutableCtxMgr(MutableTableInfo&& table);
+    std::shared_ptr<MutableCtxMgr> enter();
```

Comment:
I think we can improve on this object a bit because its a bit clunky. Some ideas:

- `MutableCtxMgr` takes a `MutableTableInfo` in the constructor, meaning the table is already locked before you call `enter()`. `enter()` should be the step that does the actual locking. Instead, if the context manager accepted a `MessageMeta`, then `enter()` could actually lock the table
- `enter()` returning a pointer to itself is also weird. It kind of seems like `enter()` doesnt do much if the table is already locked and we just return a pointer to itself. Instead, we should have another object returned here
- I know there were some issues with using RAII with context managers from python, but I still think we could benefit from using that here. We should internally store an RAII object in the python wrapper. Once that object goes out of scope, it can guarantee that the table is unlocked even if the wrapper is still alive

Reply:
I thought about making some sort of wrapped object that looks/acts like a dataframe, but instead proxies calls to the actual dataframe something like:
```
WrappedDF():
    _df
    _ctx
    
    def __call__(self, *a, **k):
        if not self._ctx.locked:
            raise RuntimeError
        else:
            # call _df
```

The problem would be anyone who needed to call some third-party lib that expected a real proper cudf dataframe and might throw an exception on our wrapped work-alike class.